pkg/actors/actor.go

```diff
@@ -11,56 +11,87 @@ import (
 	"time"
 
 	diag "github.com/dapr/dapr/pkg/diagnostics"
+	"github.com/pkg/errors"
 )
 
+var (
+	// ErrActorDisposed is the error when runtime tries to hold the lock of the disposed actor.
+	ErrActorDisposed error = errors.New("actor is already disposed")
+)
+
+// actor represents single actor object and maintains its turn-based concurrency.
 type actor struct {
+	// actorType is the type of actor.
 	actorType string
-	actorID   string
+	// actorID is the ID of actorType.
+	actorID string
+
+	// concurrencyLock is the lock to maintain actor's turn-based concurrency.
+	concurrencyLock *sync.Mutex
+	// pendingActorCalls is the number of the current pending actor calls by turn-based concurrency.
+	pendingActorCalls int32
```

Comment:
If you use https://godoc.org/go.uber.org/atomic#Int32 we are forced to use atomic operations by default and avoid future typos

Reply:
switched to atomic wrapper. thanks.